sdk/pubkey/src/lib.rs

```diff
@@ -304,6 +304,12 @@ impl Pubkey {
         Self(pubkey_array)
     }
 
+    /// Decode a string into a Pubkey, usable in a const context
+    pub const fn from_str_const(s: &str) -> Self {
```

Comment:
Hello, I'm just passing by, I think this could use a `&'static str` input and renamed to `from_static`, similar pattern as `HeaderValue::from_static`. Using `&'static` will prevent people from accidentally using this function for faillible inputs.

Reply:
I don't think this will accomplish what you think it will

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=0d6c57cbc59f7fafb6dbb22e01c720af

Reply:
I think what you're really looking for is the thing a macro does which is to check that the input is a (string) literal

Reply:
something like this, which was the previous solution

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=4b9aed32eef1d5ada53b05fd834c0a25

Reply:
I know it's not a perfect solution, but the function isn't intented to be called with a live `'a`, so there is no reason to allow it in the function signature. Using `&' static` prevent common misuse, if someone leaks a string, they should already know what they are doing. I think, as a library design principle, function signature should clearly communicate what is it use case, using `&str` tell users that it works with a live `'a`.

Reply:
In the meantime I've undone the deprecation of the pubkey macro and converted it to a declarative macro in solana-pubkey, since this discussion revealed that `Pubkey::from_str_const` does not make it completely obsolete